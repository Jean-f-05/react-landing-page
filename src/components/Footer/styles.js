import styled, { css } from 'styled-components';
import { Wrapper as FooterElement } from '../../components/TextComponent/styles';
import { Wrapper as SectionContainer } from '../../components/SectionContainer/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  text-align: center;
  border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration:none;
      font-size: ${theme.fonts.small};
      }

  & ${FooterElement} {
    font-size: ${theme.fonts.small};
}

  & ${SectionContainer}{
      padding-top: 0;
      padding-bottom: 0;
  }
`}`;
