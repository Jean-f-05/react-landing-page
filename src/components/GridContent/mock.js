export default {
  title: 'GRID CONTENT',
  html: `<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Zenonis est, inquam, hoc Stoici. Restinguet citius, si ardentem acceperit. Sed quanta sit alias, nunc tantum possitne esse tanta. Duo Reges: constructio interrete. </p>

  <h2>Similiter sensus, cum accessit ad naturam, tuetur illam quidem, sed etiam se tuetur;</h2>

  <p>Primum in nostrane potestate est, quid meminerimus? Quod iam a me expectare noli. Nihil ad rem! Ne sit sane; Sed haec omittamus; Praeteritis, inquit, gaudeo. </p>`,
  background: false,
};
