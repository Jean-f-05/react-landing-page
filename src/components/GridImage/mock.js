export default {
  background: false,
  title: 'My Grid',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?',
  grid: [
    {
      altText: 'GRID 1',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 2',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 3',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 4',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 5',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 6',
      src: 'https://source.unsplash.com/random/300x300',
    },
  ],
};
