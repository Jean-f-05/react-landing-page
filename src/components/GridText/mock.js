export default {
  background: false,
  title: 'My Grid',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?',
  grid: [
    {
      title: 'GRID 1',
      description:
        'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.',
    },
    {
      title: 'GRID 2',
      description:
        'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.',
    },
  ],
};
