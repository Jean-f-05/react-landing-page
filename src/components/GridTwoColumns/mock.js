/* eslint-disable */

export default {
    title: 'GridTwoColumns',
    text: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Veniam labore totam eaque odio mollitia rerum iusto non ex, obcaecati earum itaque tempore quo autem? Nam quam dolore asperiores tempora illum.`,
    srcImg: "assets/images/javascript.svg",
}