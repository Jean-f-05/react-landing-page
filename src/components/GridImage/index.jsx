import * as S from './styles';
import P from 'prop-types';
import SectionBackground from '../SectionBackground';
import Heading from '../Heading';
import TextComponent from '../TextComponent';

const GridImage = ({ background = false, title, description, grid }) => {
  return (
    <SectionBackground background={background}>
      <S.Wrapper>
        <Heading isUpperCase size="huge" colorDark={!background} as="h2">
          {title}
        </Heading>
        <TextComponent>{description}</TextComponent>
        <S.Grid>
          {grid.map((el) => (
            <S.GridElement key={el.altText}>
              <S.Image src={el.src} alt={el.altText} />
            </S.GridElement>
          ))}
        </S.Grid>
      </S.Wrapper>
    </SectionBackground>
  );
};

export default GridImage;

GridImage.propTypes = {
  background: P.bool,
  title: P.string.isRequired,
  description: P.string.isRequired,
  grid: P.arrayOf(
    P.shape({
      altText: P.string.isRequired,
      src: P.string.isRequired,
    }),
  ).isRequired,
};
