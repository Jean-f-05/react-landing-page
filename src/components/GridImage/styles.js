import styled, { css } from 'styled-components';
import { Header as HeadingContainer } from '../Heading/styles';
import { Wrapper as TextComponent } from '../TextComponent/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  & > ${TextComponent}{
    margin-bottom: ${theme.spacing.huge};
  }
`}`;

export const Grid = styled.div`
  ${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap:${theme.spacing.large}
`}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
`}`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(15deg);
    }
`}`;
