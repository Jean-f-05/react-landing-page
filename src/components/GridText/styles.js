import styled, { css } from 'styled-components';
import { Header as HeadingContainer } from '../Heading/styles';
import { Wrapper as TextComponent } from '../TextComponent/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  ${TextComponent}{
    margin-bottom: ${theme.spacing.huge};
  }
`}`;

export const Grid = styled.div`
  ${({ theme }) => css`
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap:${theme.spacing.large}
`}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
  ${HeadingContainer}{
    position: relative;
    left: 5rem;
  }
  ${HeadingContainer}:before{
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
`}`;
