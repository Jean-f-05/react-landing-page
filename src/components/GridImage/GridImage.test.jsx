import { renderTheme } from '../../styles/renderTheme';
import GridImage from '.';
import { screen } from '@testing-library/react';
import mock from './mock';

describe('<GridImage/>', () => {
  it('should render with background ', () => {
    const { container } = renderTheme(<GridImage {...mock}></GridImage>);
    expect(container).toMatchSnapshot();
  });

  it('should render without background ', () => {
    const { container } = renderTheme(
      <GridImage {...mock} background={undefined}></GridImage>,
    );
    expect(container).toMatchSnapshot();
  });
});
