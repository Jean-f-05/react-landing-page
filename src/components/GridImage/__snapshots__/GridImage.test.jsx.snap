// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage/> should render with background  1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 5.6rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(15deg);
  -ms-transform: scale(1.2) rotate(15deg);
  transform: scale(1.2) rotate(15deg);
}

.c0 {
  background: #fff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My Grid
        </h2>
        <p
          class="c4 c5"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?
        </p>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="GRID 1"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 2"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 3"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 4"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 5"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 6"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridImage/> should render without background  1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 5.6rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(15deg);
  -ms-transform: scale(1.2) rotate(15deg);
  transform: scale(1.2) rotate(15deg);
}

.c0 {
  background: #fff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          My Grid
        </h2>
        <p
          class="c4 c5"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?
        </p>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="GRID 1"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 2"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 3"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 4"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 5"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="GRID 6"
              class="c8"
              src="https://source.unsplash.com/random/300x300"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
